common/task/weighted_round_robin_task_scheduler.go

```diff
@@ -238,16 +233,14 @@ func (w *weightedRoundRobinTaskSchedulerImpl) dispatcher() {
 					return
 				default:
 					// if no task, don't block. Skip to next priority
-					break
+					break Submit_Loop
```

Comment:
I don't quite follow the reason for this addition?

Reply:
`break` here was breaking the `select` block but the intend was to break the loop.
Before: https://go.dev/play/p/kZoHQxnesgs
After:  https://go.dev/play/p/VBJWxNSCj7D